drivers/ksfilter/ks/connectivity.c

```diff
@@ -248,7 +248,7 @@ KspReadMediaCategory(
         return Status;
 
     /* allocate buffer for the registry key */
-    Path.Length = 0;
+    Path.Length = MediaPath.Length + GuidString.Length;
```

Comment:
This is wrong.
The `Path.Length` value is correctly updated by the `RtlAppendUnicodeStringToString` calls from below.
See also its implementation https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/rtl/unicode.c;hb=889eab78ca7c0f17a2ada9ff20de5807a293cdfe#l209

Reply:
OK, @HBelusca. I found out that this fix is indeed wrong. As it appeared, the problem is not caused by our Kernel Streaming (ks.sys) at all. The actual source of problem seems to be in PnP instead. See my new #3510 PR for details.